### ISSUE TITLE
Ringbuffer class missing from API documentation?

### ISSUE THREAD

**Issue by james-smith-za** [ska-sa/spead2#138]
https://github.com/ska-sa/spead2/blob/15124378872b3b79599c0ae1a84dddbe2eae5851/include/spead2/common_ringbuffer.h#L262-L285

I wasn't able to find this class documented anywhere on readthedocs. Was that by design?

**Comment by bmerry** [comment]
Probably just because I haven't put a lot of effort into the C++ docs. It's not needed by everybody because it's normally hidden behind `spead2::recv::ring_stream`.

**Comment by james-smith-za** [comment]
Good point. I came looking for it because of how it gets used in kafgpu, and I couldn't find anything in readthedocs. The documentation in the code was what I needed, but I just thought it was weird that Sphinx didn't include it.

**Comment by bmerry** [comment]
We can leave this bug open - one day I should get around to improving the C++ docs on readthedocs.
